api/MarbleCollectorApi/Services/UserScoreService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using MarbleCollectorApi.Data.Repository;
+using MarbleCollectorApi.ViewModels;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MarbleCollectorApi.Services
+{
+    public class UserScoreService : IUserScoreService
+    {
+        private readonly IAssignmentRepository _assignmentRepository;
+        private readonly IGrantRepository _grantRepository;
+
+        public UserScoreService(IAssignmentRepository assignmentRepository, IGrantRepository grantRepository)
+        {
+            _assignmentRepository = assignmentRepository;
+            _grantRepository = grantRepository;
+        }
+
+        public UserScore GetUserScore(int userId)
+        {
+            var choreAssignments = _assignmentRepository.GetAssignments(userId);
+            var choreAssignmentsCompleted = _assignmentRepository.GetCompletedAssignments(userId, choreAssignments); // saving a trip to the db
+            var marblesEarned = _assignmentRepository.GetMarblesEarned(userId, choreAssignmentsCompleted);
+
+            var rewardGrants = _grantRepository.GetGrants(userId);
+            var rewardGrantsCompleted = _grantRepository.GetCompletedGrants(userId, rewardGrants);
+            var marblesSpent = _grantRepository.GetMarblesSpent(userId, rewardGrants);
+
+            return new UserScore
+            {
+                Id = userId,
+                FamilyRank = 0, // can not be determined here...
+                ChoreAssignments = choreAssignments.Count(),
+                ChoreAssignmentsCompleted = choreAssignmentsCompleted.Count(),
+                RewardGrants = rewardGrants.Count(),
+                RewardsGrantsCompleted = rewardGrantsCompleted.Count(),
+                MarblesEarned = marblesEarned,
+                MarblesSpent = marblesSpent,
+                MarbleBalance = marblesEarned - marblesSpent,
```

Comment:
Irgendwie haben wir wohl komische Daten in unserer Datenbank, denn die Kinder haben alle einen negativen Kontostand an Murmeln 😁 ...
![image](https://user-images.githubusercontent.com/17404505/111691879-13164880-882f-11eb-8294-13263ac4a17e.png)
Die Berechnungen scheinen ok zu sein, aber vermutlich sind tatsächlich die Daten in der Datenbank etwas _schräg_ 😲 .

Reply:
Das ist mir auch aufgefallen, bei mir war das Kind auf dem ersten Rang mit 0 Murmeln 😄